hudi-spark-datasource/hudi-spark/src/test/scala/org/apache/hudi/functional/TestCOWDataSource.scala

```diff
@@ -619,4 +622,69 @@ class TestCOWDataSource extends HoodieClientTestBase {
       .load(basePath + "/*")
     assertTrue(recordsReadDF.filter(col("_hoodie_partition_path") =!= lit("")).count() == 0)
   }
+
+  @Test def testQueryCowWithoutStar(): Unit = {
```

Comment:
- Rename to `testQueryCOWWithBasePathAndFileIndex`
- Can we make this parameterized as well ?